### ISSUE TITLE
Music mode issues

### ISSUE THREAD

**Issue by Toufukun** [lhc70000/iina#1224]
**System and IINA version:**

- macOS 10.13.1
- IINA 0.0.14.1

**Expected behavior:**

1. IINA will close the file being played and open the file I drop on a music mode window.

2. Able to enter music mode again after I quit the music mode.

3. Cover art will be displayed in music mode.

4. Multi-track lossless audio (.cue) will show in playlists instead of chapters.

**Actual behavior:**

1. When I want to "replace" the song I am now listening to with a new song (i.e. I want to switch the song I listen to), I will drag and drop the file on the music mode window, but IINA will do nothing. I have to drop it on the icon on Dock, and IINA will open a new music mode window. I have to close the previous window by myself.

2. When I click the X button to close a music mode window, it returns to normal mode, and there is no option in menus to enter the music mode again. This confuses me very often. As I said in previous item, I have to "close" twice to close a music mode window. Please separate closing and quitting music mode into different buttons.

3. Please consider showing album cover arts in music mode.

4. IINA will treat CUEs as a long audio and its tracks will show in chapters. The seek bar has the same length as the full track too (over 1 hour long). Can you let them show in normal playlist like Foobar2000?

- [x] MPV does not have this problem.

**Comment by lhc70000** [comment]
2. Video - Music Mode (⎇⌘M)
3. It'll be added in the next version
4. No, it's mpv's logic.

**Comment by Toufukun** [comment]
@lhc70000 Thanks. For item 4, can you do a trick to convert them into tracks in playlist? I don't think it must follow mpv's logic. (In fact, I did not expect mpv can play CUEs so I never tried it.)

**Comment by mtcls** [comment]
Agree. Since Vox player become a shit, I would like to use IINA as audio player too. CUE parsing would be great feature.

**Comment by ptkovsky** [comment]
@mtcls IINA is great for local video playback, it has hotkeys and stuff. But it's little bit heavier and bulkier as an audio player in terms of resource usage and overall UX. Here is an example of the issue: in the particular case when I open my audio playlist - it doesn't even start in Music Mode. Another issue: to manually switch to Music Mode, you need to click on the Video submenu instead of Audio submenu. From the logical standpoint and design-wise, this issues already look paradoxical if you ask me. And there may be many more small usability issues likes this in this UI implementation. To be honest, in the end it's not a dedicated audio player and it is mostly designed as a universal solution after all.

As of today, on macOS I end up using DeaDBeeF or foobar2000 for audio purposes. They are both great alternatives to Vox. Earlier, somewhere around 2016, I was kind of skeptical about this projects (for starters, foobar2000 port didn't even exist till recent time), but now they are being developed with a decent pace. Check them out if you didn't yet! :)

**Comment by arigit** [comment]
+1 on cue handling + external cover art

**Comment by CarterLi** [comment]
+1 on cue handling

**Comment by grishka** [comment]
+1, I still have to use VOX to play those weird "one huge .flac + .cue if you want actual tracks" albums
